plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -13,38 +13,69 @@
  */
 package io.trino.plugin.password.ldap;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.ImmutableSet;
-import io.trino.plugin.password.Credential;
+import com.google.common.io.Closer;
 import io.trino.spi.security.AccessDeniedException;
 import io.trino.spi.security.BasicPrincipal;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
-import javax.naming.NamingException;
-
-import java.util.HashSet;
-import java.util.Optional;
-import java.util.Set;
-
+import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 
 public class TestLdapAuthenticator
 {
-    private static final String BASE_DN = "base-dn";
-    private static final String PATTERN_PREFIX = "pattern::";
+    private static final String DEFAULT_IMAGE = "bitnami/openldap:2.5.11";
+
+    private static final int LDAP_PORT = 1389;
+
+    private final Closer closer = Closer.create();
+
+    private static final String BASE_DN = "dc=example,dc=org";
+    private static final String PATTERN = "(&(objectClass=inetOrgPerson)(uid=${USER}))";
+
+    private LdapAuthenticatorClient client;
+
+    @BeforeClass
+    public void setup()
+    {
+        GenericContainer<?> openLdapContainer = new GenericContainer<>(DEFAULT_IMAGE)
```

Comment:
Please extract this as separate class `TestingLdap` so it could be reused between tests.